src/pudl/analysis/allocate_net_gen.py

```diff
@@ -514,24 +527,36 @@ def associate_generator_tables(
 ) -> pd.DataFrame:
     """Associate the three tables needed to assign net gen and fuel to generators.
 
-    The ``generation_fuel_eia923`` table's data is reported at the
-    :py:const:`IDX_PM_ESC` granularity. Each generator has one ``prime_mover_code``, but
-    potentially several ``energy_source_code``s (reported in as primary source,
-    secondary source, etc.). Much of the data in ``generation_fuel_eia923`` not
-    associated with primary energy sources, so before merging the data with the
-    generators, we employ :func:``stack_generators``.
-
-    After the generator's energy source codes have been stacked, we group the other
-    data tables and merge them with the generators. We add suffixes to the data columns
-    to identify the source table.
+    The :ref:`generation_fuel_eia923` table's data is reported at the
+    :py:const:`IDX_PM_ESC` granularity. Each generator in the :ref:`generators_eia860`
+    has one ``prime_mover_code``, but potentially several ``energy_source_code``s that
+    are reported in several columns. We need to reshape the generators energy source
+    code columns to rows so it can be merged with the :ref:`generation_fuel_eia923`
+    table. We do this using :func:``stack_generators`` employing :func:`pd.stack` which
+    creates one column of ``energy_sorce_code`` with all of the codes that are present
+    in the many code columns in the :ref:`generators_eia860` table.
+
+    The stacked generators table has primary keys of:
+    ``["plant_id_eia", "generator_id", "report_date", "energy_sorce_code"]``. Other
+    columns include the ``prime_mover_code`` column in it for merging in the other data
+    tables, the ``capacity_mw`` which we use to allocate on if there is no data in the
+    :ref:`generation_eia923` table and the ``operational_status`` which we use to remove
+    inactive plants from the association and allocation process.
```

Comment:
The stacked generators table has a primary key of ``["plant_id_eia", "generator_id", "report_date", "energy_source_code"]``. The table also includes the ``prime_mover_code`` column to enable merges with other tables, the ``capacity_mw`` column which we use to determine the allocation when there is no data in the :ref:`generation_eia923` table, and the ``operational_status`` column which we use to remove inactive plants from the association and allocation process.